ci/check_disallowed_imports.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+import collections
+import fnmatch
+import json
+import pathlib
+import subprocess
+import sys
+
+CURRENT_DIR = pathlib.Path(__file__).parent.absolute()
+
+
+def generate_dependency_graph(*args):
+    command = ("pydeps", "--nodot", "--show-deps") + args
+    print(f"Running: {' '.join(command)}")  # noqa: T201
+    result = subprocess.run(command, check=True, capture_output=True, text=True)
+    return json.loads(result.stdout)
+
+
+def check_dependency_rules(dependency_graph, disallowed_imports):
+    prohibited_deps = collections.defaultdict(set)
+
+    for module, module_data in dependency_graph.items():
+        imports = module_data.get("imports", [])
+
+        for pattern, disallow_rules in disallowed_imports.items():
+            if fnmatch.fnmatch(module, pattern):
+                for disallow_rule in disallow_rules:
+                    for imported in imports:
+                        if fnmatch.fnmatch(imported, disallow_rule):
+                            prohibited_deps[module].add(imported)
```

Comment:
```suggestion
				prohibited_deps[module] |= {
					imported
					for disallow_rule in disallow_rules
					for imported in imports
					if fnmatch.fnmatch(imported, disallow_rule)
				}
```

The nesting is getting a bit deep here 😅